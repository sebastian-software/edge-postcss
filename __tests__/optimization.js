import { compile } from "./core"


// ====================================================
// ================ OPTIMIZATION ======================
// ====================================================

test("Calc Trivial", () =>
  compile(".elem { width: calc(300px + 10px); }")
)

test("Calc Variable", () =>
  compile("$margin: 10px; .elem { width: calc(200px + $margin); }")
)

test("Calc Keep", () =>
  compile(".elem { width: calc(100px + 2%); }")
)

test("zIndex", () =>
  compile(".first { z-index: 1000; } .second { z-index: 2000; }")
)

test("CSS-O (Optimizer)", () =>
  compile(".elem { color: #ff0000; }")
)
