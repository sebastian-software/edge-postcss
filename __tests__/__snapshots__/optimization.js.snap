// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSS-O (Optimizer) 1`] = `
".elem{
color:red
}"
`;

exports[`Calc Keep 1`] = `
".elem{
width:calc(100px + 2%)
}"
`;

exports[`Calc Trivial 1`] = `
".elem{
width:310px
}"
`;

exports[`Calc Variable 1`] = `
".elem{
width:210px
}"
`;

exports[`zIndex 1`] = `
".first{
z-index:1
}

.second{
z-index:2
}"
`;
